test/runtests.jl

```diff
@@ -1,24 +1,63 @@
-using MLJText # substitute for correct interface pkg name
+using MLJText
 using Test
 using MLJBase
-import Distributions
-using StableRNGs # for RNGs stable across all julia versions
-rng = StableRNGs.StableRNG(123)
-
-@testset "cool classifier" begin
-    n = 100
-    p = 3
-    nclasses = 5
-    X, y = make_blobs(n, p, centers=nclasses, rng=rng);
-    model = MLJText.CoolProbabilisticClassifier()
-    mach = machine(model, X, y)
-    fit!(mach, rows=1:2, verbosity=0)
-    yhat = predict(mach, rows=3)
-    @test size(Distributions.pdf(yhat, levels(y)) ) == (1, nclasses)
-
-    e = evaluate!(mach, measure=BrierLoss(), verbosity=0)
-    @test e.measurement[1] < 1.0
-
-    @test fitted_params(mach).classes_seen_in_training == levels(y)
-    @test report(mach).n_classes_seen == nclasses
+using TextAnalysis
+
+@testset "tfidf transformer" begin
+    # add some test docs
+    docs = ["Hi my name is Sam.", "How are you today?"]
+
+    # convert to ngrams
+    ngram_vec = ngrams.(documents(Corpus(NGramDocument.(docs))))
+
+    # train transformer
+    tfidf_transformer = MLJText.TfidfTransformer()
+    test_machine = @test_logs machine(tfidf_transformer, ngram_vec)
+    MLJBase.fit!(test_machine)
+
+    # test
+    test_doc = ngrams(NGramDocument("Another sentence ok"))
+    test1 = transform(test_machine, [test_doc])
+    @test sum(test1, dims=2)[1] == 0.0
+    @test size(test1) == (1, 11)
+
+    test_doc2 = ngrams(NGramDocument("Listen Sam, today is not the day."))
+    test2 = transform(test_machine, [test_doc2])
+    @test sum(test2, dims=2)[1] > 0.0
+    @test size(test2) == (1, 11)
+
+    test_doc3 = ngrams.(
+        Corpus([NGramDocument("Another sentence ok"), NGramDocument("Listen Sam, today is not the day.")])
+    )
+    test3 = transform(test_machine, test_doc3)
+    @test sum(test3, dims=2)[1] == 0.0
+    @test sum(test3, dims=2)[2] > 0.0
+    @test size(test3) == (2, 11)
+
+    test_doc4 = [["Another", "sentence", "ok"], ["Listen", "Sam", ",", "today", "is", "not", "the", "day", "."]]
+    test4 = transform(test_machine, test_doc4)
+    @test sum(test4, dims=2)[1] == 0.0
+    @test sum(test4, dims=2)[2] > 0.0
+    @test size(test4) == (2, 11)
+    # test with bag of words
+    bag_of_words = Dict(
+        "cat in" => 1,
+        "the hat" => 1,
+        "the" => 2,
+        "cat" => 1,
+        "hat" => 1,
+        "in the" => 1,
+        "in" => 1,
+        "the cat" => 1
+    )
+    bag = Dict{MLJText.NGram, Int}(Tuple(String.(split(k))) => v for (k, v) in bag_of_words)
```

Comment:
```suggestion
    bag = Dict(Tuple(String.(split(k))) => v for (k, v) in bag_of_words)
```

With this change, I get tests passing on Julia 1.3. This is a reasonable test for correctness of your algorithm, no?  I don't see we need to limit the user to Julia 1.3, as the fail does indeed look like a Julia bug rather than something wrong with with the implementation. 

Thanks for diagnosing the issue. I imagine this was a bit of pain 😥 . No-one wants to see "unreachable..."!

Reply:
@pazzo83

Reply:
interesting - it looks like with the change to the methods where I'm trying to build the corpus that you suggested (the `build_corpus` methods), the type that is resolved by simply having `Dict(Tuple(String.(split(k))) => v for (k, v) in bag_of_words)` is able to be routed to the proper method.  Thanks!!